File: 卡码/代码随想录/Python/字符串/string/02-反转字符串2/LeetCode_0541.py
#541. 反转字符串 II
from typing import *

class Solution:
  def reverseStr(self, s: str, k: int) -> str:
    s=list(s)
    for i in range(0,len(s),2*k):
      start=i
      end=min(start+k-1,len(s)-1)
      while start<end:
        s[start],s[end]=s[end],s[start]
        start+=1
        end-=1
    return "".join(s)

File: 卡码/代码随想录/1/04-字符串/Js/02-反转字符串2/LeetCode_0541.js
//541. 反转字符串 II
const reverseStr = function(s, k) {
  let ch=Array.from(s);
  for(let i=0;i<ch.length;i+=2*k){
    let start=i-1,end=Math.min(i+k,ch.length);
    while(++start<--end) [ch[start],ch[end]]=[ch[end],ch[start]];
  }
  return ch.join("");
};